### PR TITLE
Qt: Make Qt separate package and add Qt6 support

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Converts Qt into its own "package" and adds necessary build files and changes for Qt6.

Also makes the Qt5 build process modular by downloading only the required submodules instead of the complete Qt package.

### Motivation and Context
Prepares for a move to Qt6 sometime in 2022/2023 as well as establish Qt as its own module in preparation of building Qt for Windows on CI as well.

### How Has This Been Tested?
* Qt5 and Qt6 build on macOS M1 host
* Cross-compiled Qt5 for x86_64 on M1 host

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
This can be rebased now that #98 has been merged.

**Comment by PatTheMav** [comment]
Tested the generated Qt6 deps with a local obs-studio build without issues.

**Comment by gxalpha** [comment]
Sure, I'll test as soon as I have a functional build environment again - as explanation, macOS 13 doesn't correctly identify one of my displays and doesn't at all finds the other one making me unable to really do any serious tasks.
Once I rolled back to 12 I'll test this PR.

File: deps.qt/checksums/qtnetworkauth-everywhere-src-5.15.2.tar.xz.sha256 [hunk 1]
```diff
@@ -0,0 +1 @@
+fcc2ec42faa68561efa8f00cd72e662fbc06563ebc6de1dc42d96bb2997acd85  qtnetworkauth-everywhere-src-5.15.2.tar.xz
```
Comment (RytoEX) [inline deps.qt/checksums/qtnetworkauth-everywhere-src-5.15.2.tar.xz.sha256:1]: I don't believe that we actually use qtnetworkauth, so you should be able to omit this.

File: deps.qt/checksums/qtnetworkauth-everywhere-src-6.3.0.tar.xz.sha256 [hunk 2]
```diff
@@ -0,0 +1 @@
+203a98942919028520038f90a20cd7ee32b537233545d11e429c3e2c1ad9069f  qtnetworkauth-everywhere-src-6.3.0.tar.xz
```
Comment (RytoEX) [inline deps.qt/checksums/qtnetworkauth-everywhere-src-6.3.0.tar.xz.sha256:1]: I don't believe that we actually use qtnetworkauth, so you should be able to omit this.

File: deps.qt/checksums/qtwinextras-everywhere-src-5.15.2.tar.xz.sha256 [hunk 3]
```diff
@@ -0,0 +1 @@
+65b8272005dec00791ab7d81ab266d1e3313a3bbd8e54e546d984cf4c4ab550e  qtwinextras-everywhere-src-5.15.2.tar.xz
```
Comment (RytoEX) [inline deps.qt/checksums/qtwinextras-everywhere-src-5.15.2.tar.xz.sha256:1]: This would be needed for the Windows build if we end up downloading the individual component archives. Currently, I'm still working against the git super repo, so you can omit this from this PR.

File: utils.zsh/30-build.zsh [hunk 4]
```diff
@@ -14,10 +14,12 @@ setup_target() {
     [output_dir]="${project_root}/${1}/obs-${PACKAGE_NAME}-${1##*-}"
   )
 
+  mkdir -p "${project_root}/${1}/obs-${PACKAGE_NAME}-${1##*-}"
+
   case ${1} {
     macos-x86_64)
       config_data+=(
-        [deployment_target]=10.13
+        [deployment_target]=10.15
```
Comment (RytoEX) [inline utils.zsh/30-build.zsh:22]: Is this change intentional/necessary?

**Review by RytoEX — CHANGES_REQUESTED** [review]
Just a quick visual review.

File: utils.zsh/30-build.zsh (hunk 4 above)
Comment (PatTheMav) [reply utils.zsh/30-build.zsh:22]: Yes, @WizardCM and I talked about raising this for 10.15 anyway with macOS 13 releasing this year.

Also Qt6 complains about a 10.13 target because it requires 10.14 at the very least.

**Comment by PatTheMav** [comment]
Removed the unnecessary checksum files and also fixed release creation job.

File: utils.zsh/30-build.zsh (hunk 4 above)
Comment (PatTheMav) [reply utils.zsh/30-build.zsh:22]: As discussed off-thread we will raise the minimum supported version to 10.15 with the next OBS release, so this aligns.

**Comment by PatTheMav** [comment]
> Overall, this looks fine to me. Do we want the relevant build steps to be named "Build macOS Qt" or "Build macOS Qt5" and "Build macOS Qt6"?

They carry the Qt version explicitly right now iirc - did I miss something?

**Comment by RytoEX** [comment]
> > Overall, this looks fine to me. Do we want the relevant build steps to be named "Build macOS Qt" or "Build macOS Qt5" and "Build macOS Qt6"?
> 
> They carry the Qt version explicitly right now iirc - did I miss something?

.github/workflows/main.yml lines 311 and 402. The build step names, not the build job names.

**Comment by PatTheMav** [comment]
> > > Overall, this looks fine to me. Do we want the relevant build steps to be named "Build macOS Qt" or "Build macOS Qt5" and "Build macOS Qt6"?
> > 
> > 
> > They carry the Qt version explicitly right now iirc - did I miss something?
> 
> .github/workflows/main.yml lines 311 and 402. The build step names, not the build job names.

Good catch, will make them explicit.

**Review by RytoEX — APPROVED** [review]
Seems fine to me. I think @gxalpha also managed to build with these artifacts.
